runtime/gc_glue_java/EnvironmentDelegate.hpp

```diff
@@ -159,6 +170,40 @@ class MM_EnvironmentDelegate
 
 	void forceOutOfLineVMAccess();
 
+	MMINLINE void
+	preMove(omrobjectptr_t objectPtr)
```

Comment:
I'd actually stick 'Object' in the name (here and in EnvBase too)
While the parameter somewhat implies that we move object, class name  (Environment) or instance name (env0 when we invoke it, does not. So, a naive thought would be that we move Env (or something inside Env).

Reply:
It would also be nice if the name is descriptive enough to say that we are using this only for _sliding_ (as opposed to _evacuating_ like copy-forward)  object movement. The former can partially damage the source object, in particular object header if sliding is less the object size (rare but possible), while the latter cannot (dest object is in dedicated survivor area). This is why we need to save those flags pre movement, since we need them later (for potential object growth).

Anyhow, what I want to say, we could stick 'Compact' in the name, too (here and EnvBase, but not in CompactScheme/WriteOnceCompactor - it's clear there).

Or if you fill it's not necessary to change the name, at least you should put a comment for methods what kind of object movement this is/is not used for.

Reply:
I'm wasting out time on something not that important, but still, I realized that compact in preObjectCompact would imply that object is being compacted (while in fact the heap is being compacted). preObjectSlide would be correct (we are indeed sliding object), but we don't use 'slide' that frequently in code/comments, and perhaps perObjectCompactMove would be ok (Compact would be noun as adjective to Move: we move object, but movement is compact style. Pick one, or come up with something better :))

Anyhow, I think we are ready to start breaking this into smaller, serialized PRs....

Reply:
That makes sense, hmm.. What about `preObjectMoveForCompact` and `postObjectMoveForCompact`?